diff --git a/tensorflow_metadata/proto/v0/BUILD b/tensorflow_metadata/proto/v0/BUILD
index 24cf416..9e426aa 100644
--- a/tensorflow_metadata/proto/v0/BUILD
+++ b/tensorflow_metadata/proto/v0/BUILD
@@ -14,8 +14,6 @@
 # ==============================================================================


-load("@com_google_protobuf//bazel:py_proto_library.bzl", "py_proto_library")
-
 licenses(["notice"])  # Apache 2.0

 package(default_visibility = ["//visibility:public"])
@@ -24,26 +22,20 @@ proto_library(
     name = "metadata_v0_proto",
     srcs = [
         "anomalies.proto",
-        "derived_feature.proto",
         "metric.proto",
         "path.proto",
         "problem_statement.proto",
         "schema.proto",
         "statistics.proto",
+        "derived_feature.proto"
     ],
     deps = [
         # For well-known proto types like protobuf.Any.
         "@com_google_protobuf//:any_proto",
-        "@com_google_protobuf//:api_proto",
-        "@com_google_protobuf//:compiler_plugin_proto",
         "@com_google_protobuf//:descriptor_proto",
         "@com_google_protobuf//:duration_proto",
-        "@com_google_protobuf//:empty_proto",
-        "@com_google_protobuf//:field_mask_proto",
-        "@com_google_protobuf//:source_context_proto",
         "@com_google_protobuf//:struct_proto",
         "@com_google_protobuf//:timestamp_proto",
-        "@com_google_protobuf//:type_proto",
         "@com_google_protobuf//:wrappers_proto",
     ],
 )
@@ -60,10 +52,12 @@ cc_library(
     deps = [":cc_metadata_v0_proto_cc"],
 )

-py_proto_library(
+alias(
     name = "metadata_v0_proto_py_pb2",
-    deps = [
-        ":metadata_v0_proto",
-    ],
+    actual = ":metadata_v0_proto",
 )

+alias(
+    name = "metadata_v0_proto_py_pb2_genproto",
+    actual = ":metadata_v0_proto",
+)
