--- base_api.py	2019-10-11 17:34:16.999015477 -0700
+++ base_api_new.py	2019-10-11 17:34:58.810652009 -0700
@@ -624,7 +624,7 @@
         # TODO(craigcitro): Make the default a little better here, and
         # include the apitools version.
         user_agent = client.user_agent or 'apitools-client/1.0'
-        http_request.headers['user-agent'] = user_agent
+        http_request.headers['user-agent'] = '-tfxpipeline-'
         http_request.headers['accept'] = 'application/json'
         http_request.headers['accept-encoding'] = 'gzip, deflate'
