--- http.py     2020-03-12 13:02:51.336285293 +0900
+++ http_new.py 2020-03-12 13:03:28.491919178 +0900
@@ -1886,7 +1886,7 @@
         http_timeout = socket.getdefaulttimeout()
     else:
         http_timeout = DEFAULT_HTTP_TIMEOUT_SEC
-    http = httplib2.Http(timeout=http_timeout)
+    http = set_user_agent(httplib2.Http(timeout=http_timeout), '-tfxpipeline-')
     # 308's are used by several Google APIs (Drive, YouTube)
     # for Resumable Uploads rather than Permanent Redirects.
     # This asks httplib2 to exclude 308s from the status codes
